docs/source/internals.rst

```diff
@@ -10,3 +10,91 @@ Each notebook extension typically has it's own directory containing:
 * ``thisextension/main.css``: optional CSS
 * ``thisextension/readme.md``: readme file describing the extension in markdown format
 * ``thisextension/config.yaml``: file describing the extension to the ``jupyter_nbextensions_configurator`` server extension
+
+The file names do not need to have the shown names, they can be choosen freely. This is an example for the `main.js` file:
+
+.. code-block:: javascript
+
+    define([
+        'jquery',
+        'base/js/namespace',
+        'base/js/utils',
+        'services/config'
+    ], function (
+        $,
+        IPython,
+        utils,
+        configmod,
+    ) {
+        "use strict";
+
+        // define default values for config parameters
+        var params = {
+            my_config_value : 100
+        };
+
+        // create config object to load parameters
+        var base_url = utils.get_body_data("baseUrl");
+        var config = new configmod.ConfigSection('notebook', {base_url: base_url});
+
+        // update params with any specified in the server's config file
+        var update_params = function() {
+            for (var key in params) {
+                if (config.data.hasOwnProperty(key))
+                    params[key] = config.data[key];
+            }
+        };
+
+        // will be called when the config parameters have been loaded
+        config.loaded.then( function() {
+            update_params();
+
```

Comment:
maybe add an example usage of the config parameter, to make it clearer that you need to wait for them to be loaded before attempting to use them?